cmd/lotus-shed/sectors.go

```diff
@@ -131,3 +132,99 @@ var terminateSectorCmd = &cli.Command{
 		return nil
 	},
 }
+
+func findPenaltyInInternalExecutions(prefix string, trace []types.ExecutionTrace) {
+	for _, im := range trace {
+		if im.Msg.To.String() == "f099" /*Burn actor*/ {
+			fmt.Printf("Estimated termination penalty: %s attoFIL\n", im.Msg.Value)
+			return
+		}
+		findPenaltyInInternalExecutions(prefix+"\t", im.Subcalls)
+	}
+}
+
+var terminateSectorPenaltyEstimationCmd = &cli.Command{
+	Name:      "termination-estimate",
+	Usage:     "Estimate the termination penalty",
+	ArgsUsage: "[sectorNum1 sectorNum2 ...]",
+	Action: func(cctx *cli.Context) error {
+		if cctx.Args().Len() < 1 {
+			return fmt.Errorf("at least one sector must be specified")
+		}
+
+		nodeApi, closer, err := lcli.GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		api, acloser, err := lcli.GetStorageMinerAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer acloser()
+
+		ctx := lcli.ReqContext(cctx)
+
+		maddr, err := api.ActorAddress(ctx)
+		if err != nil {
+			return err
+		}
+
+		mi, err := nodeApi.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+		if err != nil {
+			return err
+		}
+
+		terminationDeclarationParams := []miner2.TerminationDeclaration{}
+
+		for _, sn := range cctx.Args().Slice() {
+			sectorNum, err := strconv.ParseUint(sn, 10, 64)
+			if err != nil {
+				return fmt.Errorf("could not parse sector number: %w", err)
+			}
+
+			sectorbit := bitfield.New()
+			sectorbit.Set(sectorNum)
+
+			loca, err := nodeApi.StateSectorPartition(ctx, maddr, abi.SectorNumber(sectorNum), types.EmptyTSK)
+			if err != nil {
+				return fmt.Errorf("get state sector partition %s", err)
+			}
+
+			para := miner2.TerminationDeclaration{
+				Deadline:  loca.Deadline,
+				Partition: loca.Partition,
+				Sectors:   sectorbit,
+			}
+
+			terminationDeclarationParams = append(terminationDeclarationParams, para)
+		}
+
+		terminateSectorParams := &miner2.TerminateSectorsParams{
+			Terminations: terminationDeclarationParams,
+		}
+
+		sp, err := actors.SerializeParams(terminateSectorParams)
+		if err != nil {
+			return xerrors.Errorf("serializing params: %w", err)
+		}
+
+		msg := &types.Message{
+			From:   mi.Owner,
+			To:     maddr,
+			Method: miner.Methods.TerminateSectors,
+
+			Value:  big.Zero(),
+			Params: sp,
+		}
+
+		invocResult, err := nodeApi.StateCall(ctx, msg, types.TipSetKey{})
+		if err != nil {
+			return xerrors.Errorf("fail to state call: %w", err)
+		}
+
+		findPenaltyInInternalExecutions("\t", invocResult.ExecutionTrace.Subcalls)
```

Comment:
So... yeah. Well, so.... yeah.... termination penalties.

This will _probably_ work in most cases, but isn't actually correct. When the user calls `TerminateSectors`, we:

1. Add these terminations to a queue of terminations.
2. Process (part of) the queue of terminations.
3. If we fail to process the entire queue, we'll schedule a cron callback to process the next during the next cron tick.

This means:

1. The "burn" here can technically include burn from pending terminations.
2. If the caller tries to check too many sectors here, we may not include the entire burn.

We can probably ignore this in a first-pass implementation (almost never going to be an issue and this _is_ an estimate). But we should at least leave a comment.

We could also add a function to detect pending terminations. That should solve this.